dvc/stage_cache.py

```diff
@@ -0,0 +1,75 @@
+import os
+import json
+import hashlib
+
+from dvc.utils.fs import makedirs
+
+
+def _sha256(string):
+    return hashlib.sha256(string.encode()).hexdigest()
+
+
+def _get_hash(stage):
+    if not stage.cmd or not stage.deps or not stage.outs:
+        return None
+
+    string = _sha256(stage.cmd)
+    for dep in stage.deps:
+        if not dep.def_path or not dep.get_checksum():
+            return None
+
+        string += _sha256(dep.def_path)
+        string += _sha256(dep.get_checksum())
+
+    for out in stage.outs:
+        if not out.def_path or out.persist:
```

Comment:
Why `out.persist` is None here?

Reply:
@skshetry If there is a persistent output, we can't really cache the stage. We have the same logic in our old build cache in Stage.create().